src/utils/poll_strategy_base.cpp

```diff
@@ -89,29 +89,43 @@ void PollStrategyBase::reset_state() {
 
 }
 
-void PollStrategyBase::on_assignment(TopicPartitionList& partitions) {
+void PollStrategyBase::assign(TopicPartitionList& partitions) {
     // populate partition queues
     for (const auto& partition : partitions) {
         // get the queue associated with this partition
         partition_queues_.emplace(partition, QueueData{consumer_.get_partition_queue(partition), boost::any()});
     }
     reset_state();
+}
+
+void PollStrategyBase::revoke(const TopicPartitionList& partitions) {
+    if (partitions.empty()) {
+        //revoke everything
+        partition_queues_.clear();
+    }
+    else {
+        for (const auto &partition : partitions) {
+            // get the queue associated with this partition
+            auto toppar_it = partition_queues_.find(partition);
```

Comment:
I think you can directaly call `partition_queues_.erase(partition)` here

Reply:
True, i'm not even sure where the find comes from originally. Must have been a remnant from some other action which was applied before erasing.